shell/shared/devAccounts.js

```diff
@@ -14,112 +14,91 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-var allowDevAccounts = Meteor.settings && Meteor.settings.public &&
-                 Meteor.settings.public.allowDevAccounts;
-
-if (allowDevAccounts) {
-  if (Meteor.isServer) {
-    Meteor.methods({
-      createDevAccount: function (displayName, isAdmin) {
-        // This is a login method that creates or logs in a dev account with the given displayName
-
-        check(displayName, String);
-        isAdmin = isAdmin || false;
-        var userId;
-        var user = Meteor.users.findOne({devName: displayName});
-        if (user) {
-          userId = user._id;
-        } else {
-          userId = Accounts.insertUserDoc({ profile: { name: displayName } },
-                                          { signupKey: "devAccounts", devName: displayName, isAdmin: isAdmin });
-        }
-        // Log them in on this connection.
-        return Accounts._loginMethod(this, "createDevAccount", arguments,
-            "devAccounts", function () { return { userId: userId }; });
+if (Meteor.isServer) {
+  Meteor.methods({
+    createDevAccount: function (displayName, isAdmin) {
+      if (!allowDevAccounts()) {
```

Comment:
I hate to say this after you've done all this refactoring in this file, but I wonder if we really want dev accounts to be toggle-able at runtime. Probably, we should force the user to choose between dev and prod at install time and not allow them to easily switch since it would be easy to get wrong. E.g. someone might idly enable dev accounts just to see what they are, not reading the fine print and not realizing they just opened a security hole on their server.

Perhaps we should keep this strictly in the config file...